core/src/autogluon/core/hpo/ray_hpo.py

```diff
@@ -0,0 +1,457 @@
+        searcher = SearcherFactory.get_searcher(
+            searcher_name=searcher,
+            user_init_args=user_init_args,
+            metric=metric,
+            mode=mode
+        )
+    assert isinstance(searcher, (SearchAlgorithm, Searcher)) and searcher.__class__ in searcher_presets.values()
+    # Check supported schedulers for obj input
+    if supported_searchers is not None:
+        supported_searchers_cls = [searcher_presets[searchers] for searchers in supported_searchers]
+        if searcher.__class__ not in supported_searchers_cls:
+            logger.warning(f'{searcher.__class__} is not supported yet. Using it might behave unexpected. Supported options are {supported_searchers_cls}')
+    return searcher
+
+
+def _get_scheduler(hyperparameter_tune_kwargs: dict, supported_schedulers: Optional[List[str]]=None):
+    """Initialize scheduler object"""
+    scheduler = hyperparameter_tune_kwargs.get('scheduler')
+    user_init_args = hyperparameter_tune_kwargs.get('scheduler_init_args', dict())
+    if isinstance(scheduler, str):
+        assert scheduler in scheduler_presets, f'{scheduler} is not a valid option. Options are {scheduler_presets.keys()}'
+        # Check supported schedulers for str input
+        if supported_schedulers is not None:
+            if scheduler not in supported_schedulers:
+                logger.warning(f'{scheduler} is not supported yet. Using it might behave unexpected. Supported options are {supported_schedulers}')
+        scheduler = SchedulerFactory.get_scheduler(
+            scheduler_name=scheduler,
+            user_init_args=user_init_args,
+        )
+    assert isinstance(scheduler, TrialScheduler) and scheduler.__class__ in scheduler_presets.values()
+    # Check supported schedulers for obj input
+    if supported_schedulers is not None:
+        supported_schedulers_cls = [scheduler_presets[scheduler] for scheduler in supported_schedulers]
+        if scheduler.__class__ not in supported_schedulers_cls:
+            logger.warning(f'{scheduler.__class__} is not supported yet. Using it might behave unexpected. Supported options are {supported_schedulers_cls}')
+    return scheduler
+    
+    
+class TabularRayTuneAdapter(RayTuneAdapter):
+    
+    supported_searchers = ['random', 'bayes']
+    supported_schedulers = ['FIFO']
+    
+    def get_resources_per_trial(
+        self,
+        total_resources: dict,
+        num_samples: int,
+        minimum_cpu_per_trial: int = 1,
+        minimum_gpu_per_trial: float = 1.0,
+        model_estimate_memory_usage: Optional[int] = None,
+        **kwargs,
+        ) -> dict:
+        assert isinstance(minimum_cpu_per_trial, int) and minimum_cpu_per_trial >= 1, 'minimum_cpu_per_trial must be a integer that is larger than 0'
+        assert isinstance(minimum_gpu_per_trial, (int, float)) and minimum_gpu_per_trial > 0, 'minimum_gpu_per_trial must be a integer or float that is larger than 0'
```

Comment:
So you only specify `minimum_gpu_per_trial` if your trial requires gpu and you provided gpu in `total_resource`. Otherwise, this parameter will do nothing